core/services/wifi/WifiManager.py

```diff
@@ -87,62 +105,86 @@ def __dict_from_list(data: bytes) -> Dict[str, Any]:
 
         output = {}
         for line in raw_lines:
-            key, value = line.split(b"=")
-            output[WifiManager.__decode_escaped(key)] = WifiManager.__decode_escaped(value)
+            try:
+                key, value = line.split(b"=")
+                output[WifiManager.__decode_escaped(key)] = WifiManager.__decode_escaped(value)
+            except Exception as error:
+                raise ParseError("Failed parsing dictionary data from list.") from error
 
         return output
 
     def get_wifi_available(self) -> List[ScannedWifiNetwork]:
         """Get a dict from the wifi signals available"""
-        self.wpa.send_command_scan()
-        data, result = self.wpa.send_command_scan_results()
-        if not result:
-            raise ValueError("Failed to fetch wifi list.")
-
-        networks_list = WifiManager.__dict_from_table(data)
-        return [ScannedWifiNetwork(**network) for network in networks_list]
+        try:
+            scan_data = self.wpa.send_command_scan()
+            if b"BUSY" in scan_data:
+                raise BusyError
+            data = self.wpa.send_command_scan_results()
+            networks_list = WifiManager.__dict_from_table(data)
+            return [ScannedWifiNetwork(**network) for network in networks_list]
```

Comment:
```suggestion
            return map(ScannedWifiNetwork, networks_list)
```

Reply:
I wish it worked like that.

The correct syntax would need to be:
```python
return list(map(lambda network: ScannedWifiNetwork(**network), networks_list))
```